fix: let 'set' command fail on key not found

limaa opened the pull request:
Whenever a key is not found within a 'set' command. The command should error out instead of trying to continue execution and assign the value to a partially parsed key.

Example:
`toml set --toml-path pyproject.toml tool.ruffs.target-version py39`

The table `tool.ruffs` does not exist but the current implementation captures this exception and continues along, actually executing `tool.target-version = "py39"`.

I believe a better implementation is to not capture the exception at all and error out, this way no unintended modifications to the TOML file are performed.

Another option is to continue capturing the exception but error out using `exit(1)`.

What do you think?

mrijken commented:
Thanx, I do prefer the exit(1) variant, because the user don't need the Python traceback.

limaa commented:
I have updated to use `exit(1)` and added more information on the error message, now it prints:

    error: non-existent key 'tool.ruffs.target-version' can not be set to value 'py39'
